app/code/Magento/Cron/Model/Schedule.php

```diff
@@ -51,23 +52,34 @@ class Schedule extends \Magento\Framework\Model\AbstractModel
     private $timezoneConverter;
 
     /**
+     * @var ExpressionValidator
+     */
+    private $expressionValidator;
+
+    /**
      * @param \Magento\Framework\Model\Context $context
      * @param \Magento\Framework\Registry $registry
      * @param \Magento\Framework\Model\ResourceModel\AbstractResource $resource
      * @param \Magento\Framework\Data\Collection\AbstractDb $resourceCollection
      * @param array $data
      * @param TimezoneInterface $timezoneConverter
+     * @param ExpressionValidator $expressionValidator
      */
     public function __construct(
         \Magento\Framework\Model\Context $context,
         \Magento\Framework\Registry $registry,
         \Magento\Framework\Model\ResourceModel\AbstractResource $resource = null,
         \Magento\Framework\Data\Collection\AbstractDb $resourceCollection = null,
         array $data = [],
-        TimezoneInterface $timezoneConverter = null
+        TimezoneInterface $timezoneConverter = null,
+        ExpressionValidator $expressionValidator = null
     ) {
         parent::__construct($context, $registry, $resource, $resourceCollection, $data);
-        $this->timezoneConverter = $timezoneConverter ?: ObjectManager::getInstance()->get(TimezoneInterface::class);
+        $this->timezoneConverter = $timezoneConverter
+            ?: ObjectManager::getInstance()->get(TimezoneInterface::class);
+        $this->expressionValidator = $expressionValidator
+            ?: ObjectManager::getInstance()->get(ExpressionValidator::class);
+        $this->expressionValidator->setSchedule($this);
```

Comment:
1. Only assignment operations allowed in a constructor.
2. `$expressionValidator` is a shared service by default so such design may lead to unpredictable behavior (several instances of Schedule model may inject the same instance of the validator and override values of each other).

Reply:
You are right, it was my first approach, I'll change it.